aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/sampling/pollers/RulePoller.java

```diff
@@ -28,33 +31,42 @@
     private AWSXRay client;
     private Clock clock;
     private CentralizedManifest manifest;
+    private ScheduledExecutorService executor;
 
     public RulePoller(CentralizedManifest manifest, AWSXRay client, Clock clock) {
         this.manifest = manifest;
         this.client = client;
         this.clock = clock;
+        executor = Executors.newSingleThreadScheduledExecutor();
     }
 
     public void start() {
-        Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(() -> {
+        executor.scheduleAtFixedRate(() -> {
             try {
                 pollRule();
-            } catch (Exception ex) {
-                logger.error("Encountered error polling GetSamplingRules: ", ex);
+            } catch (Throwable t) {
+                logger.error("Encountered error polling GetSamplingRules: ", t);
+                // An Error should not be handled by the application.
+                // The executor will die and not abrupt main thread.
+                if(t instanceof Error) { throw t; }
             }
         }, 0, getJitterInterval(), TimeUnit.SECONDS);
     }
 
+    public void shutdown() {
+        executor.shutdownNow();
+    }
 
     private void pollRule() {
         Instant now = clock.instant();
 
+        logger.info("Polling sampling rules.");
```

Comment:
Should we log this message as "debug" level since this message will show every certain period of time.

Reply:
This is generated every ~5 minutes so should be fine.